src/MgvOfferTaking.sol

```diff
@@ -104,56 +111,51 @@ abstract contract MgvOfferTaking is MgvHasOffers {
               return (0, local);
             }
             index = field.firstLevel1Index();
-            field = pair.level1[index];
+            field = offerList.level1[index];
           }
           local = local.level1(field);
           index = field.firstLevel0Index(index);
-          field = pair.level0[index];
+          field = offerList.level0[index];
         }
         local = local.level0(field);
-        leaf = pair.leafs[field.firstLeafIndex(index)];
+        leaf = offerList.leafs[field.firstLeafIndex(index)];
       }
       mor.leaf = leaf;
       nextId = leaf.getNextOfferId();
     }
     return (nextId, local);
   }
-
   /* # General Market Order */
   //+clear+
   /* General market orders set up the market order with a given `taker` (`msg.sender` in the most common case). Returns `(totalGot, totalGave, penaltyReceived, feePaid)`.
   Note that the `taker` can be anyone. This is safe when `taker == msg.sender`, but `generalMarketOrder` must not be called with `taker != msg.sender` unless a security check is done after (see [`MgvOfferTakingWithPermit`](#mgvoffertakingwithpermit.sol)`. */
-  function generalMarketOrder(
-    address outbound_tkn,
-    address inbound_tkn,
-    Tick maxTick,
-    uint fillVolume,
-    bool fillWants,
-    address taker
-  ) internal returns (uint, uint, uint, uint) {
+
+  function generalMarketOrder(OLKey memory olKey, int logPrice, uint fillVolume, bool fillWants, address taker)
```

Comment:
I had expected the `int logPrice` to still be `Tick maxTick`. I am not entirely sure whether this change is needed or good - it seems to introduce some imprecision and possibly some extra calculations.

I am not sure I've found all the places where tick<->logprice conversion happens :)

Reply:
@lnist the initial idea is that it is more intuitive to specify the log of the price rather than the index of the offer in the data structure that stores it. Before this PR, tick was always equal to logPrice, so there was no choice to make between the two.

@espendk now that `snipes` are gone, we might as well say `order`? The `marketOrder` function can definitely both work a (now true) limit order and as a market order.

Reply:
> now that `snipes` are gone, we might as well say `order`? The `marketOrder` function can definitely both work a (now true) limit order and as a market order.

True. I think `order` would be clearer. And `limitOrder` even more so, since a DeFi market order in Mangrove will be modeled by adding slippage to the limit logPrice. But `order` could be a good compromise.

Reply:
> But if we retain log price, I think it should be called maxLogPrice, limitLogPrice or similar.

Fixed in 6b1891bf75d66f6c11fbe4c42d0a93ee3e490dfa